### PR TITLE
 Add support for using template variables without brackets

### PR DESCRIPTION
**Pull request by SimonBarendse**

### CODE REVIEW

File: internals/secrethub/tpl/v2.go [hunk 1]
```diff
@@ -188,8 +188,15 @@ func (p *v2Parser) parseRoot() (node, error) {
 		return variable, p.readRune()
 	}
 
-	if p.current == token.Dollar && (unicode.IsLetter(p.next) || p.next == '_') {
-		return nil, ErrUnexpectedDollar(p.lineNo, p.columnNo)
+	if p.current == token.Dollar && p.isVariableStartRune(p.next) {
+		variable, err := p.parseVarWithoutBrackets()
+		if err == io.EOF {
```
Comment (jpcoenen) [inline internals/secrethub/tpl/v2.go]: Do we need this check here?

File: internals/secrethub/tpl/v2_test.go [hunk 2]
```diff
@@ -91,6 +91,94 @@ func TestParserV2_parse(t *testing.T) {
 				},
 			},
 		},
+		"uppercase var without brackets": {
```
Comment (jpcoenen) [inline internals/secrethub/tpl/v2_test.go]: What about a variable with an underscore?

File: internals/secrethub/tpl/v2.go (hunk 1 above)
Comment (SimonBarendse) [reply internals/secrethub/tpl/v2.go]: We could also just `return p.parseVarWithoutBrackets()`

Comment (jpcoenen) [reply internals/secrethub/tpl/v2.go]: My thought exactly. Keeping it short; keeping it simple :smile:

**Pull request merged**